### DISCUSSION TITLE
STM32 Counter Callback "Busy"

### DISCUSSION THREAD

**Discussion by darshanmirajkar-eaton** [zephyrproject-rtos/zephyr#69252] [Q&A]
referring to https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/counter/counter_ll_stm32_timer.c in Ln  280

![image](https://github.com/zephyrproject-rtos/zephyr/assets/118359137/ee910cb5-9d2a-4a5a-9efa-af3b2aced20f)

Is there any specific reason to keep this callback busy??

**Comment by erwango** [comment]
Yes, see API https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/drivers/counter.h#L450
Note there is a `counter_cancel_channel_alarm` function which you can invoke if needed.
